packages/math/src/pool/concentrated/math.ts

```diff
@@ -296,43 +296,84 @@ export function convertTokenInGivenOutToTokenOutGivenIn(
 // add liquidity
 // docs ref: https://github.com/osmosis-labs/osmosis/blob/b764323ce7702185d2089b9e76a0115c7058f37e/x/concentrated-liquidity/README.md#L573
 
+// calcAmount0
 export function calculateDepositAmountForQuote(
-  curSqrtPrice: Dec,
-  lowerTick: Int,
   upperTick: Int,
-  baseDeposit: Int
-): Int {
-  const sqrtPu = tickToSqrtPrice(upperTick);
-  const sqrtPl = tickToSqrtPrice(lowerTick);
-  const sqrtPc = curSqrtPrice;
-
-  // calculate liquidity needed for token0
-  const L = new Dec(baseDeposit)
-    .mul(sqrtPu)
-    .mul(sqrtPl)
-    .quo(sqrtPu.sub(sqrtPl));
-
-  // calculate delta y
-  const deltaY = L.mul(sqrtPc.sub(sqrtPl));
-
-  return deltaY.truncate();
+  currentSqrtPrice: Dec,
+  baseDeposit: Dec
+): Dec {
```

Comment:
Could we try to use Dec for all amounts in the function signature (here: baseDeposit, and the return value)

It's fine to convert to dec then truncate to convert back. It's just standard to calculate on amounts with Ints.

Reply:
Did you mean try to use Ints for all amounts?

Reply:
Done here https://github.com/osmosis-labs/osmosis-frontend/pull/1629/commits/3bcfbff73aca5e0e02304b253bbbe44d3132d21f

It also fixed the 49.999 percent issue!